mlflow/genai/scorers/__init__.py

```diff
@@ -17,24 +74,24 @@
         Safety,
         get_all_scorers,
     )
-else:
-    # Lazy imports to avoid circular dependencies at runtime
-    def __getattr__(name):
-        if name in [
-            "Correctness",
-            "ExpectationsGuidelines",
-            "Guidelines",
-            "RelevanceToQuery",
-            "RetrievalGroundedness",
-            "RetrievalRelevance",
-            "RetrievalSufficiency",
-            "Safety",
-            "get_all_scorers",
-        ]:
-            from mlflow.genai.scorers import builtin_scorers
-
-            return getattr(builtin_scorers, name)
-        raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
+
+# Explicitly declare module-level attributes for Sphinx autodoc.
+# This __annotations__ dict helps documentation generators (like Sphinx) understand
+# what attributes are available in this module without having to execute __getattr__.
+# It provides type information for each lazily-loaded scorer class.
+__annotations__ = {
+    "Correctness": "type[mlflow.genai.scorers.builtin_scorers.Correctness]",
+    "ExpectationsGuidelines": "type[mlflow.genai.scorers.builtin_scorers.ExpectationsGuidelines]",
+    "Guidelines": "type[mlflow.genai.scorers.builtin_scorers.Guidelines]",
+    "RelevanceToQuery": "type[mlflow.genai.scorers.builtin_scorers.RelevanceToQuery]",
+    "RetrievalGroundedness": "type[mlflow.genai.scorers.builtin_scorers.RetrievalGroundedness]",
+    "RetrievalRelevance": "type[mlflow.genai.scorers.builtin_scorers.RetrievalRelevance]",
+    "RetrievalSufficiency": "type[mlflow.genai.scorers.builtin_scorers.RetrievalSufficiency]",
+    "Safety": "type[mlflow.genai.scorers.builtin_scorers.Safety]",
+    "get_all_scorers": (
+        "Callable[[], dict[str, type[mlflow.genai.scorers.builtin_scorers.BuiltInScorer]]]"
+    ),
+}
```

Comment:
Can we rely on `getattr` instead so that we don't have to keep these types updated? They can very easily become out of date if we move things around because `.builtin_scorers` is not a public module.

Alternatively, if attempting to move these without updating `__annotations__` results in a clear CI failure, that's okay too, but trying to understand why we don't just let Sphinx call `getattr`